### PR TITLE
Fix: correctly pass api_base and add --max-run-time CLI argument in main_full_repo

### PR DESCRIPTION
**Pull request by cuso4huang**
### What this PR does

This PR fixes two issues encountered when running the `main_full_repo` entry point:

1. **`api_base` not passed to `AICaller`:**  
   The CLI argument `--api_base` was accepted but never passed into `AICaller`, making it ineffective.  
   ✅ Fixed by modifying the initialization of `AICaller` to include `api_base`.

3. **Missing `--max-run-time` argument:**  
   The CLI argument parser `parse_args_full_repo` did not include `--max-run-time`, which led to runtime errors like:  
'Namespace' object has no attribute 'max_run_time'
   ✅ Added the missing argument to `argparse`.

Optionally, I also added `max_tokens` manually to the `AICaller` init call with a default value (`8192`) to avoid instantiation errors.

---

### Suggested labels:
`bug`, `CLI`, `enhancement`

Let me know if any changes are needed — happy to update!

### CODE REVIEW

File: cover_agent/utils.py
Comment (AdamWalkerQodo) [inline cover_agent/utils.py:1]: hello

Comment (cuso4huang) [reply cover_agent/utils.py:1]: Hi! Let me know if anything needs to be adjusted :)

**Comment by EmbeddedDevops1** [comment]
@cuso4huang Thank you for putting this PR and issue together. Are you able to post a sample run output from the command line?

**Comment by cuso4huang** [comment]
> @cuso4huang Thank you for putting this PR and issue together. Are you able to post a sample run output from the command line (including the command that you used to run this)?

Hi,this is the result

```bash
(qodo) cuso4@cuso4 qodo-cover (main) $ poetry run cover-agent-full-repo \
  --project-language="python" \
  --project-root="/home/cuso4/qodo-cover/templated_tests/python_fastapi" \
  --code-coverage-report-path="/home/cuso4/qodo-cover/templated_tests/python_fastapi/coverage.xml" \
  --test-command="coverage run -m pytest /home/cuso4/qodo-cover/templated_tests/python_fastapi/tests --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term" \
  --model="openai/deepseek-chat" \
  --api-base="https://api.deepseek.com"
============
Test files to be extended:
/home/cuso4/qodo-cover/templated_tests/python_fastapi/tests/test_app.py
============


Initializing language server...
LSP server initialized.
/home/cuso4/miniconda3/envs/qodo/lib/python3.11/site-packages/tree_sitter/__init__.py:36: FutureWarning: Language(path, name) is deprecated. Use Language(ptr, name) instead.
  warn("{} is deprecated. Use {} instead.".format(old, new), FutureWarning)
Context files for test file '/home/cuso4/qodo-cover/templated_tests/python_fastapi/tests/test_app.py':
/home/cuso4/qodo-cover/templated_tests/python_fastapi/app/app.py


Analyzing test file against context files...
Printing results from LLM model...
is_this_a_unit_test: 1
main_file: |
  app/app.py
```
Test file: `/home/cuso4/qodo-cover/templated_tests/python_fastapi/tests/test_app.py`,
is a unit test file for source file: `/home/cuso4/qodo-cover/templated_tests/python_fastapi/app/app.py`
Converting test command: `coverage run -m pytest /home/cuso4/qodo-cover/templated_tests/python_fastapi/tests --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term`
 to run only a single test: `coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term`
Streaming results from LLM model...
language: python
testing_framework: pytest
number_of_tests: 2
test_headers_indentation: 0
```

Streaming results from LLM model...
language: python
testing_framework: pytest
number_of_tests: 2
relevant_line_number_to_insert_tests_after: 29
relevant_line_number_to_insert_imports_after: 5
```

2025-04-24 09:38:39,672 - cover_agent.UnitTestValidator - INFO - Running build/test command to generate coverage report: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:38:42,413 - cover_agent.UnitTestValidator - INFO - Initial coverage: 63.64%
2025-04-24 09:38:42,415 - cover_agent.CoverAgent - INFO - Current Coverage: 63.64%
2025-04-24 09:38:42,415 - cover_agent.CoverAgent - INFO - Desired Coverage: 70%
Streaming results from LLM model...
language: python
existing_test_function_signature: |
  def test_add(num1, num2, expected):
new_tests:
- test_behavior: |
    Test that the current_date endpoint returns today's date in ISO format
  lines_to_cover: |
    [23]
  test_name: |
    test_current_date
  test_code: |
    def test_current_date():
        """Test that the current_date endpoint returns today's date"""
        response = client.get("/current-date")
        assert response.status_code == 200
        assert response.json() == {"date": date.today().isoformat()}
  new_imports_code: |
    ""
  test_tags: happy path
- test_behavior: |
    Test that the subtract endpoint correctly subtracts two numbers
  lines_to_cover: |
    [46]
  test_name: |
    test_subtract
  test_code: |
    @pytest.mark.parametrize(
        "num1,num2,expected",
        [
            (5, 3, 2),
            (-5, 3, -8),
            (0, 0, 0),
            (100, 50, 50),
        ]
    )
    def test_subtract(num1, num2, expected):
        """Test that the subtract endpoint works correctly"""
        response = client.get(f"/subtract/{num1}/{num2}")
        assert response.status_code == 200
        assert response.json() == {"result": expected}
  new_imports_code: |
    ""
  test_tags: happy path
- test_behavior: |
    Test that the divide endpoint raises an HTTPException when dividing by zero
  lines_to_cover: |
    [76, 77]
  test_name: |
    test_divide_by_zero
  test_code: |
    def test_divide_by_zero():
        """Test that dividing by zero raises an HTTPException"""
        response = client.get("/divide/5/0")
        assert response.status_code == 400
        assert response.json() == {"detail": "Cannot divide by zero"}
  new_imports_code: |
    ""
  test_tags: edge case
- test_behavior: |
    Test that the square endpoint correctly calculates the square of a number
  lines_to_cover: |
    [86]
  test_name: |
    test_square
  test_code: |
    @pytest.mark.parametrize(
        "number,expected",
        [
            (5, 25),
            (-3, 9),
            (0, 0),
            (10, 100),
        ]
    )
    def test_square(number, expected):
        """Test that the square endpoint works correctly"""
        response = client.get(f"/square/{number}")
        assert response.status_code == 200
        assert response.json() == {"result": expected}
  new_imports_code: |
    ""
  test_tags: happy path
```

2025-04-24 09:39:14,698 - cover_agent.UnitTestValidator - INFO - Running test with the following command: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:39:17,062 - cover_agent.UnitTestValidator - INFO - Test passed and coverage increased. Current coverage: 65.91%
2025-04-24 09:39:17,063 - cover_agent.UnitTestValidator - INFO - Running test with the following command: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:39:18,941 - cover_agent.UnitTestValidator - INFO - Test passed and coverage increased. Current coverage: 68.18%
2025-04-24 09:39:18,942 - cover_agent.UnitTestValidator - INFO - Running test with the following command: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:39:20,539 - cover_agent.UnitTestValidator - INFO - Test passed and coverage increased. Current coverage: 72.73%
2025-04-24 09:39:20,539 - cover_agent.UnitTestValidator - INFO - Running test with the following command: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:39:22,066 - cover_agent.UnitTestValidator - INFO - Test passed and coverage increased. Current coverage: 75.0%
2025-04-24 09:39:22,104 - cover_agent.UnitTestValidator - INFO - Running build/test command to generate coverage report: "coverage run -m pytest tests/test_app.py --cov=/home/cuso4/qodo-cover/templated_tests/python_fastapi/app --cov-report=xml --cov-report=term"
2025-04-24 09:39:23,714 - cover_agent.UnitTestValidator - INFO - Initial coverage: 75.0%
2025-04-24 09:39:23,714 - cover_agent.CoverAgent - INFO - Reached above target coverage of 70% (Current Coverage: 75.0%) in 0 iterations.
2025-04-24 09:39:23,714 - cover_agent.CoverAgent - INFO - Total number of input tokens used for LLM model openai/deepseek-chat: 3601
2025-04-24 09:39:23,714 - cover_agent.CoverAgent - INFO - Total number of output tokens used for LLM model openai/deepseek-chat: 657
(qodo) cuso4@cuso4 qodo-cover (main) $ 
```

**Comment by EmbeddedDevops1** [comment]
@cuso4huang we're ready to merge but I see there are conflicts. Can you rebase and resolve please?

**Comment by EmbeddedDevops1** [comment]
Sorry @cuso4huang can you rebase once more? You’re next in the queue.

File: cover_agent/utils.py [hunk 1]
```diff
@@ -332,11 +332,17 @@ def parse_args_full_repo():
...
+    "--max-run-time",
+    type=int,
+    default=30,
+    help="Maximum time (in seconds) allowed for test execution. Overrides the value in configuration.toml if provided. Defaults to 30 sec",
+)
+parser.add_argument(
+    "--record-mode",
+    action="store_true",
+    help="Enable record mode for LLM responses. Default: False.",
+)
```
Comment (EmbeddedDevops1) [inline cover_agent/utils.py:345]: Looks like there is an indent issue.

**Comment by EmbeddedDevops1** [comment]
@cuso4huang Thanks for the rebase. I am noticing that the unit tests are failing. Did you take a look after rebasing?
